assets/js/block-extensions/core-query.js

```diff
@@ -0,0 +1,273 @@
+			setAttributes({ query: updatedQuery });
+			lastQueryRef.current = updatedQuery;
+		}
+	}, [updatedQuery, setAttributes]);
+
+	const relationshipsControlLabel = __('Relationship', 'tenup-content-connect');
+	const relationshipsControlHelp = __(
+		'Select a relationship to determine how related entities are retrieved.',
+		'tenup-content-connect',
+	);
+	const sourcePostControlHelp = __(
+		'Choose the post from which related entities will be pulled. Defaults to the current post.',
+		'tenup-content-connect',
+	);
+
+	const resetAll = () => {
+		setAttributes({
+			showRelated: false,
+			sourcePost: undefined,
+			relationshipKey: undefined,
+			orderByRelationship: true,
+		});
+	};
+
+	const onShowRelatedChange = (value) => {
+		if (!value) {
+			resetAll();
+		} else {
+			setAttributes({
+				showRelated: value,
+				sourcePost: [
+					{
+						id: currentPostId,
+						type: currentPostType,
+						uuid: uuidv4(),
+					},
+				],
+			});
+		}
+	};
+
+	const onSourcePostChange = (ids) => {
+		setAttributes({ sourcePost: ids.length ? ids : undefined });
+	};
+
+	const onRelationshipChange = (value) => {
+		setAttributes({ relationshipKey: value });
+	};
+
+	const onOrderByRelationshipChange = (value) => {
+		setAttributes({ orderByRelationship: value });
+	};
+
+	return (
+		<InspectorControls>
+			<ToolsPanel label={__('Related', 'tenup-content-connect')} resetAll={resetAll}>
+				<ToolsPanelItem
+					hasValue={() => !!showRelated}
+					label={__('Related entities', 'tenup-content-connect')}
+					onDeselect={() => resetAll()}
+					isShownByDefault
+				>
+					<ToggleControl
+						label={__('Only show related entities', 'tenup-content-connect')}
+						checked={showRelated}
+						onChange={onShowRelatedChange}
+					/>
+				</ToolsPanelItem>
+				{showRelated && (
+					<ToolsPanelItem
+						hasValue={() => !!sourcePost}
+						label={__('Source post', 'tenup-content-connect')}
+						onDeselect={() => setAttributes({ sourcePost: undefined })}
+						isShownByDefault
+					>
+						<BaseControl help={sourcePostControlHelp}>
+							<ContentPicker
+								onPickChange={onSourcePostChange}
+								mode="post"
+								content={sourcePost}
+								contentTypes={postTypesSlugs}
+								maxContentItems={1}
+								singlePickedLabel={__('Selected post:', 'tenup-content-connect')}
+							/>
+						</BaseControl>
+					</ToolsPanelItem>
+				)}
+				{showRelated && (
+					<ToolsPanelItem
+						hasValue={() => !!relationshipKey}
+						label={__('Relationship', 'tenup-content-connect')}
+						onDeselect={() => setAttributes({ relationshipKey: undefined })}
+						isShownByDefault
+					>
+						{hasRelationships && (
+							<SelectControl
+								options={relationshipsOptions}
+								value={relationshipKey}
+								label={relationshipsControlLabel}
+								onChange={onRelationshipChange}
+								help={relationshipsControlHelp}
+								__nextHasNoMarginBottom
```

Comment:
```suggestion
								__nextHasNoMarginBottom
								__next40pxDefaultSize
```

If we need this control we should make sure it has the new default size to make it look better :)